cadquery/hull.py

```diff
@@ -292,24 +302,36 @@ def arc_arc(a1: Arc, a2: Arc) -> Tuple[float, Segment]:
         Segment(Point(x12, y12), Point(x22, y22)),
     )
 
-    return angles[ix], segments[ix]
+    segment_angle = atan2p(segments[ix].b.x - xc2, segments[ix].b.y - yc2)
+    if (segment_angle < a2.a1 or segment_angle > a2.a2):
+        return inf, Segment(Point(inf, inf), Point(inf, inf))
 
+    return angles[ix], segments[ix]
 
-def get_angle(current: Entity, e: Entity) -> Tuple[float, Segment]:
 
-    if current is e:
+def get_angle(angle: float, current: Entity, e: Entity) -> Tuple[float, Segment]:
+    def center(e: Entity):
+        if isinstance(e, Point):
+            return e
+        return e.c
+    if center(current) == center(e):
+        if (isinstance(current, Arc) and isinstance(e, Arc)):
+            if (current.s == e.e and not e.s == current.e):
+                return pt_pt(current.e, e.s)
+            if (e.s == current.e and not current.s == e.e):
+                return (current.a2 + pi) % (2 * pi), Segment(Point(inf, inf), Point(inf, inf))
```

Comment:
These changes allow a circle that's broken into multiple arcs continue around the circle